src/types.ts

```diff
@@ -6,6 +6,25 @@ export enum BaseType {
   RangeBase = 'RANGE_BASE'
 }
 
+/**
+ * Option types are parameter types for create functions.
+ */
+export interface HashBaseOptions {
+  name: string
+  prefixSize: number
+}
+export interface CountBaseOptions {
+  name: string
+  bucketSize: number
+}
+export interface RangeBaseOptions<RangeKey, IdKey> {
```

Comment:
`RangeBaseOptions<RangeKey, IdKey>` should be `RangeBaseOptions<RangeKey extends string, IdKey extends string>`